test/e2e/framework.go

```diff
@@ -161,6 +161,21 @@ func (f *Framework) afterEach() {
 		summaries = append(summaries, f.logsSizeVerifier.GetSummary())
 	}
 
+	if testContext.GatherMetricsAfterTest {
+		// TODO: enable Scheduler and ControllerManager metrics grabbing when Master's Kubelet will be registered.
+		grabber, err := metrics.NewMetricsGrabber(f.Client, true, false, false, true)
+		if err != nil {
+			Logf("Failed to create MetricsGrabber. Skipping metrics gathering.")
+		} else {
+			received, err := grabber.Grab(nil)
+			if err != nil {
+				Logf("MetricsGrabber failed grab metrics. Skipping metrics gathering.")
+			} else {
+				summaries = append(summaries, (*metricsForE2E)(&received))
```

Comment:
I don't understand this line. You're casting `received` to `metricsForE2E`. But shouldn't you also call something like `PrintHumanReadable` on it?